include/logger/sink.hpp

```diff
@@ -0,0 +1,71 @@
+#ifndef SINK_HPP
+#define SINK_HPP
+
+#include "formatter.hpp"
+
+namespace logger {
+class ISink
+{
+public:
+  /**
+   * @brief Assigns the pointer to the formatter.
+   * @param formatter_ptr to format the data.
+   * @retval None
+   */
+  ISink(std::shared_ptr<logger::IFormatter> &formatter_ptr);
+
+  /**
+   * @brief To access the record method.
+   * @param msg log message
+   * @param level log level
+   * @param logger_name logger name 
+   * @param log_credentials extracted log credentials
+   * @retval bool
+   */
+  bool i_record(std::string msg,
```

Comment:
Do not add i_ all public function must start with Capital letter and all private can start with small letter